src/ErrorProne.NET.CoreAnalyzers/CoreAnalyzers/SetContainsAnalyzer.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using ErrorProne.NET.Core;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace ErrorProne.NET.CoreAnalyzers
+{
+    /// <summary>
+    /// An analyzer that warns about incorrect usage of concurrent collection types.
+    /// </summary>
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public sealed class SetContainsAnalyzer : DiagnosticAnalyzerBase
+    {
+        /// <nodoc />
+        public static DiagnosticDescriptor Rule => DiagnosticDescriptors.EPC23;
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        /// <nodoc />
+        public SetContainsAnalyzer()
+            : base(Rule)
+        {
+        }
+
+        /// <inheritdoc />
+        protected override void InitializeCore(AnalysisContext context)
+        {
+            context.RegisterOperationAction(AnalyzeInvocationOperation, OperationKind.Invocation);
+        }
+
+        private void AnalyzeInvocationOperation(OperationAnalysisContext context)
+        {
+            try
+            {
+                // Detecting the calls to concurrentDictionaryInstance.OrderBy because
+                // it may fail with ArgumentException if the collection is being mutated concurrently at the same time.
```

Comment:
Seems like comment was copied from `ConcurrentCollectionAnalyzer` and does not apply here.

Reply:
True!